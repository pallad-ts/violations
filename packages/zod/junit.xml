<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="jest tests" tests="2" failures="0" errors="0" time="0.722">
  <testsuite name="extractViolationFromZodError" errors="0" failures="0" skipped="0" timestamp="2024-01-14T16:04:55" time="0.694" tests="2">
    <testcase classname="extractViolationFromZodError should return the all issues for input: { foo: 100 }" name="extractViolationFromZodError should return the all issues for input: { foo: 100 }" time="0.002">
    </testcase>
    <testcase classname="extractViolationFromZodError should return the all issues for input: { foo: &apos;bar&apos; }" name="extractViolationFromZodError should return the all issues for input: { foo: &apos;bar&apos; }" time="0.001">
    </testcase>
  </testsuite>
</testsuites>