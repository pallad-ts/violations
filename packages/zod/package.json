{
  "name": "@pallad/violations-zod",
  "version": "1.0.0",
  "main": "compiled/index.js",
  "compiled": "compiled/index.d.ts",
  "scripts": {
    "test": "pallad-scripts test",
    "compile": "pallad-scripts compile",
    "prepack": "npm run generate-barrels",
    "prepublishOnly": "npm run compile",
    "lint": "pallad-scripts lint",
    "generate-barrels": "barrelsby --delete -d ./src"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/pallad-ts/violations.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "Łukasz Kużyński <lukasz.kuzynski@gmail.com> (http://wookieb.pl)",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/pallad-ts/violations/issues"
  },
  "homepage": "https://github.com/pallad-ts/violations#readme",
  "dependencies": {
    "@pallad/violations": "^1.3.1",
    "@sweet-monads/either": "^3.3.1",
    "zod": "^3.22.4"
  }
}
