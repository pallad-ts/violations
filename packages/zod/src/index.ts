/**
 * @file Automatically generated by barrelsby.
 */

export * from "./convertZodErrorToViolationsList.ts";
export * from "./extractViolationsFromZodError";
export * from "./fromZodValidation";
export * from "./fromZodValidationAndKnownInput";
