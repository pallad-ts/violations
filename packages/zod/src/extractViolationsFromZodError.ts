import {z} from 'zod';
import {createViolation} from "@pallad/violations";

export function extractViolationsFromZodError(error: z.ZodError) {
	return Array.from(violationsFromZodError(error));
}

function* violationsFromZodError(error: z.ZodError) {
	const flattenErrors = error.flatten();
	for (const [field, errors] of Object.entries(flattenErrors.fieldErrors)) {
		if (errors) {
			for (const error of errors) {
				yield createViolation(error, field);
			}
		}
	}

	for (const error of flattenErrors.formErrors) {
		yield createViolation(error);
	}
}
