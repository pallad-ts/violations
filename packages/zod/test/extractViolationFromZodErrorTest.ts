import {z} from 'zod';
import {extractViolationsFromZodError} from "@src/extractViolationsFromZodError";

describe('extractViolationFromZodError', () => {

	const schema = z.object({
		foo: z.string(),
	}).transform((value, ctx) => {
		if (value.foo !== 'foo') {
			ctx.addIssue({
				code: z.ZodIssueCode.custom,
				message: 'foo',
			});
			return z.NEVER;
		}
		return value;
	});


	it.each([
		[{foo: 100}],
		[{foo: 'bar'}]
	])('should return the all issues for input: %s', (input) => {
		const result = schema.safeParse(input);

		if (!result.success) {
			expect(extractViolationsFromZodError(result.error)).toMatchSnapshot();
		}
	})
})
