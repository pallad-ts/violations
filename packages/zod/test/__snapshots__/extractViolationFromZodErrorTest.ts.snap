// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractViolationFromZodError should return the all issues for input: { foo: 'bar' } 1`] = `
[
  {
    "code": undefined,
    "message": "foo",
    "params": undefined,
    "path": undefined,
  },
]
`;

exports[`extractViolationFromZodError should return the all issues for input: { foo: 100 } 1`] = `
[
  {
    "code": undefined,
    "message": "Expected string, received number",
    "params": undefined,
    "path": [
      "foo",
    ],
  },
]
`;
